src/Functions/seriesOutliersDetectTukey.cpp

```diff
@@ -0,0 +1,283 @@
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunction.h>
+#include <cmath>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int BAD_ARGUMENTS;
+extern const int ILLEGAL_COLUMN;
+}
+
+///Detects a possible anomaly in series using [Tukey Fences](https://en.wikipedia.org/wiki/Outlier#Tukey%27s_fences)
+class FunctionSeriesOutliersDetectTukey : public IFunction
+{
+public:
+    static constexpr auto name = "seriesOutliersDetectTukey";
+
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionSeriesOutliersDetectTukey>(); }
+
+    std::string getName() const override { return name; }
+
+    bool isVariadic() const override { return true; }
+
+    size_t getNumberOfArguments() const override { return 0; }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+    {
+        FunctionArgumentDescriptors mandatory_args{{"time_series", &isArray<IDataType>, nullptr, "Array"}};
+        FunctionArgumentDescriptors optional_args{
+            {"kind", &isString<IDataType>, isColumnConst, "const String"},
+            {"min_percentile", &isNativeNumber<IDataType>, isColumnConst, "Number"},
+            {"max_percentile", &isNativeNumber<IDataType>, isColumnConst, "Number"},
+            {"k", &isNativeNumber<IDataType>, isColumnConst, "Number"}};
+
```

Comment:
As all arguments in l. 40 - 44 are optional, I believe that this creates unwanted overloads:

- `seriesOutliersDetectTukey(time_series, kind)`
- `seriesOutliersDetectTukey(time_series, kind, min_percentile)`
- `seriesOutliersDetectTukey(time_series, kind, min_percentile, max_percentile)`

(please check)

Unfortunately, the `FunctionArgumentDescriptors` helper class isn't great at handling this. What you might do as alternative is to either
- more work: rewrite the `getReturnTypeImpl()` code to not use `FunctionArgumentDescriptors`, or
- less work: add a check that the number of arguments (`arguments`, l. 39) is either 1 or 5

EDIT: I see that the implementation truly allows these overloads (l. 75, l. 85). Really, that's overkill, imho. Two overloads are just fine (the ones I suggested in my comment to the markdown file).